third_party/packages/ansi_up/lib/ansi_up.dart

```diff
@@ -2,76 +2,316 @@
+      //  }
+      //  var match = this._buffer.match(this._osc_regex);
+      //  if (match === null) {
+      //    pkt.kind = PacketKind.ESC;
+      //    pkt.text = this._buffer.slice(0, 1);
+      //    this._buffer = this._buffer.slice(1);
+      //    return pkt;
+      //  }
+      //  pkt.kind = PacketKind.OSCURL;
+      //  pkt.url = match[1];
+      //  pkt.text = match[2];
+      //  var rpos = match[0].length;
+      //  this._buffer = this._buffer.slice(rpos);
+      //  return pkt;
+      //}
+    }
+    return pkt;
+  }
 
-@JS()
-class TextWithAttr {
-  external AU_Color get fg;
-  external AU_Color get bg;
-  external bool get bold;
-  external String get text;
-}
+  void _processAnsi(TextPacket textPacket) {
+    final sgrCmds = textPacket.text.split(';');
+    while (sgrCmds.isNotEmpty) {
+      final sgrCmdStr = sgrCmds.removeAt(0);
```

Comment:
Add a TODO to change 
sgrCmds.removeAt(0) to tracking
an index in sgrCmds. That should be more efficient in Dart.